contracts/feirari/AutoRewardsDistributor.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity ^0.8.0;
+
+import "./../staking/ITribalChief.sol";
+import "./RewardsDistributorAdmin.sol";
+import "../refs/CoreRef.sol";
+import "./IRewardsDistributorAdmin.sol";
+
+/// @notice Controller Contract to set tribe per block in Rewards Distributor Admin on Rari
+contract AutoRewardsDistributor is CoreRef {
+    /// @notice rewards distributor admin address
+    IRewardsDistributorAdmin public rewardsDistributorAdmin;
+    /// @notice tribal chief contract address
+    ITribalChief public tribalChief;
+    /// @notice address of the CToken this contract controls rewards for
+    address public cTokenAddress;
+    /// @notice boolean which decides the action to incentivize
+    bool public isBorrowIncentivized;
+    /// @notice reward index on tribal chief to grab this staked token wrapper's index
+    uint256 public tribalChiefRewardIndex;
+
+    event SpeedChanged(bool borrowSpeedChanged, uint256 newSpeed, address cToken);
+
+    /// @notice constructor function
+    /// @param coreAddress address of core contract
+    /// @param _rewardsDistributorAdmin address rewards distributor admin contract
+    /// @param _isBorrowIncentivized boolean to incentivize borrow or supply, but not both
+    /// @param _tribalChiefRewardIndex uint of index for this contract's rewards in tribalchief
+    /// @param _cTokenAddress address of ctoken contract
+    /// @param _tribalChief address of tribalchief contract
+    constructor(
+        address coreAddress,
+        IRewardsDistributorAdmin _rewardsDistributorAdmin,
+        ITribalChief _tribalChief,
+        uint256 _tribalChiefRewardIndex,
+        address _cTokenAddress,
+        bool _isBorrowIncentivized
+    ) CoreRef(coreAddress) {
+        isBorrowIncentivized = _isBorrowIncentivized;
+        cTokenAddress = _cTokenAddress;
+        tribalChiefRewardIndex = _tribalChiefRewardIndex;
+        rewardsDistributorAdmin = _rewardsDistributorAdmin;
+        tribalChief = _tribalChief;
+    }
+
+    function _deriveRequiredCompSpeed() private view returns (uint256 compSpeed) {
+        /// @notice call the TribalChief and grab the current allocation points for the stakedTokenWrapper
+        (,,, uint120 allocPoints,) = tribalChief.poolInfo(tribalChiefRewardIndex);
+        /// @notice get the total allocation points in the TribalChief
+        uint256 totalAllocPoints = tribalChief.totalAllocPoint();
+        /// @notice get the amount of tribe being distributed on a per block basis in the TribalChief
+        uint256 tribePerBlock = tribalChief.tribePerBlock();
+        /// @notice calculate the amount per block that the Rari rewards distributor is receiving from the TribalChief
+        compSpeed = (tribePerBlock * allocPoints) / totalAllocPoints;
```

Comment:
If for some reason totalAllocPoints is zero (unlikely, but possible), this will throw a divide by zero error.
Checking this first as a short-circuit should fix it.

Reply:
fixed